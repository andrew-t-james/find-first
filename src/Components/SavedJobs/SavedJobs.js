import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

export const SavedJobs = ({ jobs }) => {
  const renderJobs = jobs.map(job =>
    <Link to={`/saved-jobs/${job.id}`} key={job.id} >
      <Card {...job} key={job.id}/>
    </Link>
  );

  return (
    <section className="cards-section saved-card-section">
      {jobs.length ? renderJobs : null}
    </section>
  );
};

export default SavedJobs;

SavedJobs.propTypes = {
  jobs: PropTypes.arrayOf(PropTypes.object)
};