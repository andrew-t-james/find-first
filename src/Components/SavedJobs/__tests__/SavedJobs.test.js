import React from 'react';
import { shallow } from 'enzyme';
import { SavedJobs } from '../SavedJobs';

describe('<SavedJobs />', () => {
  let wrapper;
  const mockJobs = [{title: 'job-1', id: 1}, {title: 'job-2', id: 2}];

  beforeEach(() => wrapper = shallow(<SavedJobs jobs={mockJobs} />));

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
