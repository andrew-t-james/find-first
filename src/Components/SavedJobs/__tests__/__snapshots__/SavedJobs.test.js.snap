// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SavedJobs /> should match snapshot 1`] = `
<section
  className="cards-section saved-card-section"
>
  <Link
    key="1"
    replace={false}
    to="/saved-jobs/1"
  >
    <Card
      id={1}
      key="1"
      title="job-1"
    />
  </Link>
  <Link
    key="2"
    replace={false}
    to="/saved-jobs/2"
  >
    <Card
      id={2}
      key="2"
      title="job-2"
    />
  </Link>
</section>
`;
